app/src/main/java/com/ably/tracking/demo/subscriber/ui/screen/main/map/MainScreenMap.kt

```diff
@@ -0,0 +1,97 @@
+package com.ably.tracking.demo.subscriber.ui.screen.main.map
+
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.SideEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import com.ably.tracking.demo.subscriber.ui.screen.main.MainViewModel
+import com.ably.tracking.demo.subscriber.ui.theme.AATSubscriberDemoTheme
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.CameraPositionState
+import com.google.maps.android.compose.Circle
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.rememberCameraPositionState
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.launch
+
+const val longAnimationDuration = 1000
+const val shortAnimationDuration = 100
+const val mapZoomLevel = 16F
+
+@Composable
+fun MainScreenMap(
+    viewModel: MainViewModel
+) = AATSubscriberDemoTheme {
+    val mapState = viewModel.mapState.collectAsState().value
+    val cameraPositionState = rememberCameraPositionState()
+    val coroutineScope = rememberCoroutineScope()
+
+    /**
+     * This is a hack that solves animation stutters
+     * See https://github.com/ably/ably-asset-tracking-android/issues/680
+     * and https://github.com/ably/ably-asset-tracking-android/issues/681
+     */
+    SideEffect {
+        coroutineScope.launch {
+            when {
+                /**
+                 * THis handles the initial state of map, when it's not zoomed to anything,
+                 * and not already zooming, it should zoom to the trackable location,
+                 * before we start animating between the provided camera positions
+                 */
+                !mapState.isZoomedInToTrackable && mapState.location != null && !cameraPositionState.isMoving -> {
```

Comment:
Could this be extracted to a function like `isInInitialState`?

Reply:
I've moved it to `mapState` in https://github.com/ably/asset-tracking-android-customer-app-demo/pull/4/commits/0eb869d508157fe3024164f2a30ed4c6572cca19